src/lib.rs

```diff
@@ -183,17 +181,23 @@ pub fn hash_with_result<P: AsRef<[u8]>>(password: P, cost: u32) -> BcryptResult<
 pub fn hash_with_salt<P: AsRef<[u8]>>(
     password: P,
     cost: u32,
-    salt: &[u8],
+    salt: [u8; 16],
 ) -> BcryptResult<HashParts> {
     _hash_password(password.as_ref(), cost, salt)
 }
 
 /// Verify that a password is equivalent to the hash provided
 #[cfg(any(feature = "alloc", feature = "std"))]
 pub fn verify<P: AsRef<[u8]>>(password: P, hash: &str) -> BcryptResult<bool> {
+    use std::convert::TryInto;
+
     let parts = split_hash(hash)?;
     let salt = base64::decode_config(&parts.salt, base64::BCRYPT)?;
-    let generated = _hash_password(password.as_ref(), parts.cost, &salt)?;
+    let generated = _hash_password(
+        password.as_ref(),
+        parts.cost,
+        salt.try_into().expect("salt should be length 16"),
```

Comment:
Won’t this panic if the salt is the wrong length?

Reply:
Good catch! I'll be publishing a fix.

Reply:
The reason I used `expect` was due to an assumption that a panic would have occurred if a slice of length different from 16 would be passed through the previous implementation.